src/core/main.c

```diff
@@ -752,7 +753,14 @@ static void set_manager_defaults(Manager *m) {
         m->default_restart_usec = arg_default_restart_usec;
         m->default_start_limit_interval = arg_default_start_limit_interval;
         m->default_start_limit_burst = arg_default_start_limit_burst;
-        m->default_cpu_accounting = arg_default_cpu_accounting;
+
+        /* On 4.15+ with unified hierarchy, CPU accounting is essentially free as it doesn't require the CPU
+         * controller to be enabled, so the default is to enable it. */
+        if (need_cpu_controller_for_accounting())
+                m->default_cpu_accounting = arg_default_cpu_accounting;
+        else
+                m->default_cpu_accounting = true;
+
```

Comment:
Hmm, this overrides the user's configuration unconditionally as we parsed the config file earlier, and now forcibly override it with true

I fear the fix is a big more complicated: I figure a nice way out is to make arg_cpu_accounting an "int" instead of a "bool", and make it take < 0 for "don't know", == 0 for "no" and > 0 for "yes". We call these "tristate" at various places, and use them at a lot of places, for example we already have parsers for that in `config_parse_tristate()`...

The idea would then to make just `arg_cpu_accounting` one of these tristates, and then initialize it statically to -1. Then, in set_manager_defaults() we'd honour the setting if it is set to == 0 or > 0 as it is, but if it is still -1, we'd set it to true/false depending on whether cpu accounting is free or not... 

That way people could still turn off cpu account if they want (not sure why they would, given it is free, but whatevs...)

All the other accounting bools should remain bools, and the cpu_accounting bool inside of the Manager structure should too, only `arg_cpu_accounting` should become a tristate...